src/accelerate/data_loader.py

```diff
@@ -845,12 +845,10 @@ def prepare_data_loader(
             if hasattr(dataloader.batch_sampler, "batch_size"):
                 batch_size_for_check = dataloader.batch_sampler.batch_size
             else:
-                logger.warning(
-                    f"Make sure that you are using a correct batch size for `split_batches`. "
-                    f"We can not check it since dataloader.batch_size is None and your batch_sampler "
-                    f"{type(dataloader.batch_sampler)} does not have `batch_size` attribute."
+                raise ValueError(
+                    "Can not check batch_size for `split_batches` mode since dataloader.batch_size is None and "
+                    f"your batch_sampler {type(dataloader.batch_sampler)} does not have `batch_size` attribute."
```

Comment:
Excellent. A small mod proposal to define it more rigorously:

```suggestion
                raise ValueError(
                    "In order to use `split_batches==True` you must have a `batch_size` attribute either in the passed `dataloader` or `dataloader.batch_sampler` objects, and it has to return a natural number. Your batch sampler `{type(dataloader.batch_sampler)}` does not have the `batch_size` attribute set."
```

Reply:
And as I wrote in the error you probably also want to check that it doesn't return 0, right?

so probably need to add that check as well?

Reply:
If you are referring to `batch_size % num_process == 0`, we don't need to add it because it will be checked later by line 853.

Reply:
It won't check for == 0 because it does:

`if batch_size_for_check > 1 and batch_size_for_check % num_processes != 0:`

But we can leave it as is for now, it's unlikely it'd be set to that